store/sql_post_store.go

```diff
@@ -212,6 +212,58 @@ func (s SqlPostStore) GetFlaggedPosts(userId string, offset int, limit int) Stor
 	return storeChannel
 }
 
+func (s SqlPostStore) GetFlaggedPostsForTeam(userId, teamId string, offset int, limit int) StoreChannel {
+	storeChannel := make(StoreChannel, 1)
+	go func() {
+		result := StoreResult{}
+		pl := model.NewPostList()
+
+		var posts []*model.Post
+
+		query := `
```

Comment:
This looks rather complicated. Could there be performance concerns here? Would it make it any better if it was switched to be like
```SQL
SELECT
    *
FROM
     Posts
WHERE
    Id IN ...
    AND ChannelId IN (SELECT Id from Channels WHERE TeamId = :TeamId)
    AND DeleteAt = 0
...
```
? I guess that doesn't help in the number of subqueries, but it does remove the inner join and reduces the depth of the nesting